File: packages/core/frontend/src/types/resources/pdb.ts
export interface PodDisruptionBudget {
  Name: string;
  Namespace: string;
  MinAvailable: string;
  MaxUnavailable: string;
  CurrentHealthy: number;
  DesiredHealthy: number;
  Age: string;
}

export interface PodDisruptionBudgetDetail {
  Name: string;
  Namespace: string;
  MinAvailable: string;
  MaxUnavailable: string;
  CurrentHealthy: number;
  DesiredHealthy: number;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  Selector: Record<string, string>;
}

File: packages/core/kube/dto/pdb.go
package dto

type PodDisruptionBudgetRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type PodDisruptionBudget struct {
	Name           string
	Namespace      string
	MinAvailable   string
	MaxUnavailable string
	CurrentHealthy int32
	DesiredHealthy int32
	Age            string
}

type PodDisruptionBudgetDetail struct {
	Name           string
	Namespace      string
	MinAvailable   string
	MaxUnavailable string
	CurrentHealthy int32
	DesiredHealthy int32
	Age            string
	CreatedAt      string
	Labels         map[string]string
	Annotations    map[string]string
	Selector       map[string]string
}
